Firestore/Protos/nanopb_cpp_generator.py

```diff
@@ -188,16 +189,96 @@ def nanopb_parse_files(request, options):
     options: The command-line options from nanopb_parse_options.
 
   Returns:
-    A dictionary of filename to nanopb.ProtoFile objects, each one representing
-    the parsed form of a FileDescriptor in the request.
+    - a dictionary of filename to nanopb.ProtoFile objects, each one representing
+    the parsed form of a FileDescriptor in the request;
+    - a dictionary of filename (without `.proto` extension) to a
+      `FilePrettyPrintingInfo` object, which contains information on how to
+      generate pretty-printing code for this file.
   """
-  # Process any include files first, in order to have them
-  # available as dependencies
+  # Process any include files first, in order to have them available as
+  # dependencies
   parsed_files = {}
+  pretty_printing_info = {}
   for fdesc in request.proto_file:
-    parsed_files[fdesc.name] = nanopb.parse_file(fdesc.name, fdesc, options)
+    parsed_file = nanopb.parse_file(fdesc.name, fdesc, options)
+    parsed_files[fdesc.name] = parsed_file
 
-  return parsed_files
+    base_filename = fdesc.name.replace('.proto', '')
+    pretty_printing_info[base_filename] = FilePrettyPrintingInfo(parsed_file,
+                                                                 base_filename)
+
+  return parsed_files, pretty_printing_info
+
+
+class FilePrettyPrintingInfo:
```

Comment:
I don't particularly like the names of these classes, it's more that I used them as sort of placeholders and then got used to them. Any renaming suggestions are very welcome.

Reply:
Thanks for the suggestions! While both `ToString` and `Pprint` seem reasonable, I still slightly prefer `PrettyPrinting` to either of them. However, dropping `Generator` seems like a win, and goes a long way towards solving the verbosity problem.